lib/yard/cli/yardoc.rb

```diff
@@ -294,7 +295,16 @@ def parse_arguments(*args)
         # Last minute modifications
         self.files = Parser::SourceParser::DEFAULT_PATH_GLOB if files.empty?
         files.delete_if {|x| x =~ /\A\s*\Z/ } # remove empty ones
-        readme = Dir.glob('README{,*[^~]}').first
+        readmes = Dir.glob('README{,*[^~]}')
+        if readmes.empty?
+          readme = Dir.glob(files.first).first if options.onefile
+        else
+          readme = readmes.map {|file| Pathname file }.
```

Comment:
Please use `File.extname` instead of pulling in Pathname, since it is [significantly slower](https://github.com/lsegal/yard/blob/master/benchmarks/pathname_vs_string.rb).

Reply:
That's an outdated benchmark. Pathname is now C-accelerated, so nothing is faster than Pathname operations.

Reply:
(And that's compounded by the fact that we do many more operations than just a single extname call).

Reply:
I just did a benchmark to verify my statement. As it turns out, File.extname is still sightly faster assuming you already have a Pathname.

```
        File.extname:  2751417.2 i/s
    Pathname#extname:  2423775.1 i/s - 1.14x  slower
```

But that's not the only operation we are doing here. Let me see how the whole chain compares.

Reply:
Though I still hold to my original statement that Pathname is no longer as slow as it once was because it's now C-accelerated. We have to be careful not to carry the fallacies about Pathname forward.

Reply:
Despite the improvements to Pathname, it looks like not using Pathname is still faster. If we really want it to be as accelerated as possible, we wouldn't even use File.

```rb
readmes.
  sort {|a, b| a.slice(a.rindex('.') || a.length, a.length) <=> b.slice(b.rindex('.') || b.length, b.length) }.
  sort {|a, b| a.slice(0, a.rindex('.') || a.length) <=> b.slice(0, b.rindex('.') || b.length) }[0]
```

Reply:
The benchmark file in the repo is still reporting similar timings in Ruby 2.5.x:

```
λ ruby pathname_vs_string.rb
join:
Rehearsal -------------------------------------------------------
pathname              0.015000   0.000000   0.015000 (  0.021441)
string                0.000000   0.000000   0.000000 (  0.000626)
pathname-sameobject   0.032000   0.000000   0.032000 (  0.023407)
string-sameobject     0.000000   0.000000   0.000000 (  0.000630)
---------------------------------------------- total: 0.047000sec

                          user     system      total        real
pathname              0.015000   0.000000   0.015000 (  0.020978)
string                0.000000   0.000000   0.000000 (  0.000567)
pathname-sameobject   0.031000   0.000000   0.031000 (  0.020176)
string-sameobject     0.000000   0.000000   0.000000 (  0.000591)


cleanpath:
Rehearsal --------------------------------------------
pathname   0.000000   0.000000   0.000000 (  0.005434)
string     0.000000   0.000000   0.000000 (  0.001758)
----------------------------------- total: 0.000000sec

               user     system      total        real
pathname   0.000000   0.000000   0.000000 (  0.005791)
string     0.000000   0.000000   0.000000 (  0.001719)
```

The cost is in unnecessary allocations, not the C acceleration. The problem will get worse as you have more Pathname objects to create. There's still a valid concern here of bringing in an extra lib for a single Ruby method call that's well represented in the core functions. 

I don't think using slice is necessary. `File.extname` is a reasonable balance.

Reply:
:+1:

Reply:
I'll update the PR accordingly.